unit_test/main_tests/main_platform_test.py

```diff
@@ -68,19 +68,24 @@ def test_platform_environment(platform, intercepted_build_args, monkeypatch):
 
 
 def test_archs_default(platform, intercepted_build_args, monkeypatch):
-    monkeypatch.setattr(platform_module, 'machine', lambda: 'x86_64')
+    monkeypatch.setattr(platform_module, 'machine', lambda: 'AMD64' if platform == 'windows' else 'x86_64')
 
     main()
     build_options = intercepted_build_args.args[0]
 
     if platform == 'linux':
         assert build_options.architectures == {Architecture.x86_64, Architecture.i686}
+    elif platform == 'windows':
+        assert build_options.architectures == {Architecture.AMD64, Architecture.x86}
     else:
         assert build_options.architectures == {Architecture.x86_64}
 
 
 @pytest.mark.parametrize('use_env_var', [False, True])
 def test_archs_argument(platform, intercepted_build_args, monkeypatch, use_env_var):
+    if platform == 'windows':
+        pytest.skip('Will have empty build selectors on Windows')
```

Comment:
I'm confused. I would have expected this to be `if platform != 'linux':` Why does this test pass on macOS? There shouldn't be any selectors active when `archs` is `ppc64le` on macOS?

Reply:
macOS does not take the arch into account at all, it's not even passed. So it currently completly ignores the arch when making build selectors (until the Universal2 patch, anyway).

See:
```python
  elif platform == 'macos':
      python_configurations = cibuildwheel.macos.get_python_configurations(build_selector)
```

Reply:
Should the arch check protect against any unsupported arch, though? Why is this passing that?

Reply:
Ah, that happens in the build function. So if we move this before the build attempt, then the empty selector check happens before the architecture check, and you won't get the nicer message for that check. I think, since they are all the same, we should pull the arch check out and place it with the selector check (it gets repeated 3x anyway).